app/db/repositories/base.py

```diff
@@ -0,0 +1,56 @@
+from sqlalchemy.orm import Session
+from typing import Type, TypeVar, Generic, Optional, List
+
+T = TypeVar("T")
+
+
+class BaseRepository(Generic[T]):
+    def __init__(self, session: Session, model_class: Type[T]) -> None:
+        self._session: Session = session
+        self._model_class: Type[T] = model_class
+
+    def get_all(self) -> List[T]:
+        return self._session.query(self._model_class).all()
+
+    def get_by_id(self, item_id: int) -> Optional[T]:
+        return (
+            self._session.query(self._model_class)
+            .filter(self._model_class.id == item_id)
+            .one_or_none()
+        )
+
+    def create(self, item: T) -> T:
+        try:
+            self._session.add(item)
+            self._session.commit()
+            self._session.refresh(item)
+            return item
+        except Exception as e:
+            self._session.rollback()
+            raise e
+
+    def update(self, item: T) -> Optional[T]:
+        existing_item = self.get_by_id(item.id)
+        if existing_item:
```

Comment:
You can remove nesting in this if-body if you change condition and return None immediately.
```suggestion
        if not existing_item:
             return None
```